### ISSUE TITLE
collections, inheritance and composition

### ISSUE THREAD

**Issue by andrea-bistacchi** [gecos-lab/PZero#82]
After what we discussed in the call on June 25th, 2024, I realized that maybe we can avoid multiple inheritance in the base superclass used to derive collections by using composition. This can be done if the QAbstractTableModel becomes a property of the [BaseCollection](https://github.com/gecos-lab/PZero/blob/851f269563dd94b64df156218cd993c37aa23222/pzero/collections/AbstractCollection.py#L22C22-L22C41), instead of being its parent class.

This is already done for another fundamental component of collections: [the DataFrame](https://github.com/gecos-lab/PZero/blob/851f269563dd94b64df156218cd993c37aa23222/pzero/collections/AbstractCollection.py#L189).

```
from pandas import DataFrame

[...]

    @property
    def df(self) -> DataFrame:
        """
        Get or set the dataframe of the Collection
        """

        return self._df

    @df.setter
    def df(self, df: DataFrame):
        self._df = df
```

This should not force us to rewrite a lot of code. I guess that if QAbstractTableModel  becomes a property like `collection.qatb`, what we need is just to add `.qatb` where needed. Correct?

The discussion is open!

**Comment by andrea-bistacchi** [comment]
Or maybe also this proxy stuff, that is required to have sorting on the table view columns, could go in the collection?

**Comment by gbene** [comment]
Yes we could set the property qtab to directly return the proxy_coll and set it in self.GeologyTableView.setModel. I will try it now!

**Comment by gbene** [comment]
Hi so I finally figured out how to make it work and now it should be working with this last push https://github.com/gecos-lab/PZero/commit/d9fb18913eab51795769051524e2a50f62486592 (note: I am working from a new class_refactoring branch derived from windows_refactoring). 

While doing this I have also modified the selected_uids property in the project_window and simplified a bit the code. Before we were doing:


```python
        if self.shown_table == "tabGeology":
            selected_idxs_proxy = self.GeologyTableView.selectionModel().selectedRows()
            for idx_proxy in selected_idxs_proxy:
                selected_idxs.append(self.proxy_geol_coll.mapToSource(idx_proxy))
            for idx in selected_idxs:
                selected_uids.append(
                    self.geol_coll.data(index=idx, role=Qt.DisplayRole)
```

however this is a bit redundant and could be compacted to:

```python
        if self.shown_table == "tabGeology":
            selected_idxs_proxy = self.GeologyTableView.selectionModel().selectedRows(column=0)  # this will always give rows that have selected the column 0 (in this case uid). By changing the column=0 to another index it will give the value in another column
            for idx_proxy in selected_idxs_proxy:
                selected_uids.append(idx_proxy.data())
```

**Comment by andrea-bistacchi** [comment]
Seems solved and verified after push https://github.com/gecos-lab/PZero/commit/864eecf600e9cc8dc262a665e06b92e7ffa7f93c
